management/src/main/java/io/micronaut/management/endpoint/env/EnvironmentEndpoint.java

```diff
@@ -17,104 +17,125 @@
 
 import io.micronaut.context.env.Environment;
 import io.micronaut.context.env.PropertySource;
+import io.micronaut.core.annotation.Nullable;
 import io.micronaut.management.endpoint.annotation.Endpoint;
 import io.micronaut.management.endpoint.annotation.Read;
 import io.micronaut.management.endpoint.annotation.Selector;
+import jakarta.inject.Inject;
 
+import java.security.Principal;
 import java.util.ArrayList;
-import java.util.Arrays;
 import java.util.Collection;
 import java.util.Comparator;
 import java.util.LinkedHashMap;
-import java.util.List;
 import java.util.Map;
-import java.util.regex.Pattern;
-import java.util.stream.Collectors;
 
 /**
  * {@link Endpoint} that displays information about the environment and its property sources.
  *
  * @author Álvaro Sánchez-Mariscal
  * @since 1.2.0
  */
-@Endpoint(EnvironmentEndpoint.NAME)
+@Endpoint(
+        id = EnvironmentEndpoint.NAME,
+        defaultEnabled = EnvironmentEndpoint.DEFAULT_ENABLED
+)
 public class EnvironmentEndpoint {
 
     /**
      * Endpoint name.
      */
     public static final String NAME = "env";
 
-    public static final String[] PROPERTY_NAMES_TO_MASK = new String[] {
-            "password", "credential", "certificate", "key", "secret", "token"
-    };
+    /**
+     * Endpoint default enabled.
+     */
+    public static final boolean DEFAULT_ENABLED = false;
 
     private final Environment environment;
-    private final List<Pattern> maskPatterns;
+    private final EnvironmentEndpointFilter environmentFilter;
 
     /**
      * @param environment The {@link Environment}
      */
     public EnvironmentEndpoint(Environment environment) {
+        this(environment, null);
+    }
+
+    /**
+     * @param environment The {@link Environment}
+     * @param environmentFilter The registered {@link EnvironmentEndpointFilter} bean if one is registered
+     */
+    @Inject
+    public EnvironmentEndpoint(Environment environment, @Nullable EnvironmentEndpointFilter environmentFilter) {
         this.environment = environment;
-        this.maskPatterns = Arrays.stream(PROPERTY_NAMES_TO_MASK)
-                .map(s -> Pattern.compile(".*" + s + ".*", Pattern.CASE_INSENSITIVE))
-                .collect(Collectors.toList());
+        this.environmentFilter = environmentFilter;
     }
 
     /**
+     * @param principal The current {@link Principal} if one exists
      * @return The environment information as a map with the following keys: activeEnvironments, packages and
      * propertySources.
      */
     @Read
-    public Map<String, Object> getEnvironmentInfo() {
+    public Map<String, Object> getEnvironmentInfo(@Nullable Principal principal) {
```

Comment:
This is breaking binary compatibility. Add another deprecated method with the original signature.

Reply:
Done

Reply:
I would rather we just not handle the principal rather than adding a new method. Even adding the method is a breaking change because the original method and any extensions of it would never be called